File: src/main/java/com/skybay888/service/RouteMapService.java
package com.skybay888.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skybay888.domain.RouteMap;
import com.skybay888.dto.RouteMapDTO;
import com.skybay888.dto.RouteMapSearchDTO;
import com.skybay888.dto.RouteMapPageDTO;
import com.skybay888.dto.RouteMapConvertCriteriaDTO;
import com.skybay888.service.GenericService;
import com.skybay888.dto.common.RequestDTO;
import com.skybay888.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface RouteMapService extends GenericService<RouteMap, Integer> {

	List<RouteMap> findAll();

	ResultDTO addRouteMap(RouteMapDTO routeMapDTO, RequestDTO requestDTO);

	ResultDTO updateRouteMap(RouteMapDTO routeMapDTO, RequestDTO requestDTO);

    Page<RouteMap> getAllRouteMaps(Pageable pageable);

    Page<RouteMap> getAllRouteMaps(Specification<RouteMap> spec, Pageable pageable);

	ResponseEntity<RouteMapPageDTO> getRouteMaps(RouteMapSearchDTO routeMapSearchDTO);
	
	List<RouteMapDTO> convertRouteMapsToRouteMapDTOs(List<RouteMap> routeMaps, RouteMapConvertCriteriaDTO convertCriteria);

	RouteMapDTO getRouteMapDTOById(Integer routeMapId);







}

File: skybay888_ui/src/services/RouteMapService.js
import http from "../http-common"; 

class RouteMapService {
  getAllRouteMaps(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/routeMap/routeMaps`, searchDTO);
  }

  get(routeMapId) {
    return this.getRequest(`/routeMap/${routeMapId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/routeMap?field=${matchData}`, null);
  }

  addRouteMap(data) {
    return http.post("/routeMap/addRouteMap", data);
  }

  update(data) {
  	return http.post("/routeMap/updateRouteMap", data);
  }
  
  uploadImage(data,routeMapId) {
  	return http.postForm("/routeMap/uploadImage/"+routeMapId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new RouteMapService();
